src/DensityEquationTerms.jl

```diff
@@ -0,0 +1,120 @@
+├── grid: 1×1×100 RectilinearGrid{Float64, Flat, Flat, Bounded} on CPU with 0×0×3 halo
+├── timestepper: QuasiAdamsBashforth2TimeStepper
+├── tracers: (T, S)
+├── closure: Nothing
+├── buoyancy: SeawaterBuoyancy with g=9.80665 and BoussinesqEquationOfState{Float64} with ĝ = NegativeZDirection()
+└── coriolis: Nothing
+
+julia> GravitationalPotentialEnergy(model)
+KernelFunctionOperation at (Center, Center, Center)
+├── grid: 1×1×100 RectilinearGrid{Float64, Flat, Flat, Bounded} on CPU with 0×0×3 halo
+├── kernel_function: gravitational_potential_energy_ccc (generic function with 1 method)
+└── arguments: ("KernelFunctionOperation at (Center, Center, Center)", "KernelFunctionOperation at (Center, Center, Center)", "Tuple{Float64}")
+```
+
+To use a reference density pass the argument `reference_geopotential_height` to
+`GravitationalPotentialEnergy`:
+```jldoctest
+julia> using Oceananigans, SeawaterPolynomials.TEOS10;
+
+julia> using Oceanostics.DensityEquationTerms: GravitationalPotentialEnergy;
+
+julia> grid = RectilinearGrid(size=100, z=(-1000, 0), topology=(Flat, Flat, Bounded));
+
+julia> tracers = (:T, :S);
+
+julia> eos = TEOS10EquationOfState();
+
+julia> buoyancy = SeawaterBuoyancy(equation_of_state=eos);
+
+julia> model = NonhydrostaticModel(; grid, buoyancy, tracers);
+
+julia> reference_geopotential_height = 0; # density variable will be σ₀
+
+julia> GravitationalPotentialEnergy(model, reference_geopotential_height)
+KernelFunctionOperation at (Center, Center, Center)
+├── grid: 1×1×100 RectilinearGrid{Float64, Flat, Flat, Bounded} on CPU with 0×0×3 halo
+├── kernel_function: gravitational_potential_energy_ccc (generic function with 1 method)
+└── arguments: ("KernelFunctionOperation at (Center, Center, Center)", "KernelFunctionOperation at (Center, Center, Center)", "Tuple{Float64}")
+```
+"""
+@inline function GravitationalPotentialEnergy(model; location = (Center, Center, Center))
+
+    validate_location(location, "GravitationalPotentialEnergy")
+    validate_buoyancy(model.buoyancy)
+
+    grid = model.grid
+    ρ = seawater_density(model) # in-situ model density
+    Z = model_geopotential_height(model)
+    g = tuple(model.buoyancy.model.gravitational_acceleration)
+
+    return KernelFunctionOperation{Center, Center, Center}(gravitational_potential_energy_ccc, grid, ρ, Z, g)
+end
+@inline function GravitationalPotentialEnergy(model, reference_geopotential_height; location = (Center, Center, Center))
+
+    validate_location(location, "GravitationalPotentialEnergy")
+    validate_buoyancy(model.buoyancy)
+
+    grid = model.grid
+    σ = seawater_density(model, geopotential_height = reference_geopotential_height) # potential density reference to `reference_geopotential_height`
+    Z = model_geopotential_height(model)
+    g = tuple(model.buoyancy.model.gravitational_acceleration)
+
+    return KernelFunctionOperation{Center, Center, Center}(gravitational_potential_energy_ccc, grid, σ, Z, g)
+end
```

Comment:
Can we merge these two into one function? Would something like the suggestion below work?

```suggestion
@inline function GravitationalPotentialEnergy(model, reference_geopotential_height=nothing; location = (Center, Center, Center))

    validate_location(location, "GravitationalPotentialEnergy")
    validate_buoyancy(model.buoyancy)

    grid = model.grid
    σ = seawater_density(model, geopotential_height = reference_geopotential_height) # potential density reference to `reference_geopotential_height`
    Z = model_geopotential_height(model)
    g = tuple(model.buoyancy.model.gravitational_acceleration)

    return KernelFunctionOperation{Center, Center, Center}(gravitational_potential_energy_ccc, grid, σ, Z, g)
end
```

Reply:
Yes we can. One reason I did not is that I want to use the default behaviour of [`seawater_density`](https://clima.github.io/OceananigansDocumentation/dev/appendix/library/#Oceananigans.Models.seawater_density-Tuple{Union{HydrostaticFreeSurfaceModel,%20NonhydrostaticModel}}) for the in-situ density, and passing `nothing` to `seawater_density`,

```julia
seawater_density(model; geopotential_height  = reference_geopotential_height) # i.e.seawater_density(model; geopotential_height = nothing)
```

will not work. Instead we could use a conditional

```julia
ρ = isnothing(reference_geopotential_height) ? seawater_density(model) : seawater_density(model, geopotential_height = reference_geopotential_height) 
```

and then just need one function. Would that be ok?

Reply:
I ended up using your suggestion of adding the keyword argument 

```julia
@inline function PotentialEnergy(model; geopotential_height = model_geopotential_height(model), location = (Center, Center, Center))

    validate_location(location, "PotentialEnergy")
    validate_buoyancy(model.buoyancy)

    grid = model.grid
    ρ = seawater_density(model; geopotential_height)
    Z = model_geopotential_height(model)
    parameters = (g = model.buoyancy.model.gravitational_acceleration,)

    return KernelFunctionOperation{Center, Center, Center}(gρz_ccc, grid, ρ, Z, parameters)
end
```

The default behviour is still to use in-situ density in the calculation but setting `geopotential_height = value` in  `PotentialEnergy` will instead use a potential density reference to `value`.